src/python/pants/util/hash.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
```

Comment:
I think I recommend having this be in `docker/util_rules`. It seems unlikely that we'll want this code to be used in other parts of the codebase without careful thought. If we do want to use it in other places, we can move it to this more generic place. For now, avoid "premature generalization".

Reply:
Alright :)

Reply:
How about in the `docker/utils.py` file, as it's rather small?

Reply:
Also, I tend to want to only put files with rules in them in `docker/util_rules/..`, otherwise I find the name misleading.. ;)

Reply:
That works! And yeah...we put `interpreter_constraints` in `backend/python/util_rules` and it's a lie - there are no rules for it. But at this point it would be disruptive to change.